llvm/lib/Target/AIE/AIEPostSelectOptimize.cpp

```diff
@@ -438,6 +440,161 @@ bool duplicateAdressingRegs(MachineBasicBlock &MBB, MachineRegisterInfo &MRI) {
   return tryToDuplicateLoadUse(LoadUses, NonLoadUses, MRI, TII);
 }
 
+using Operation = AIEBaseInstrInfo::AbstractOp::OperationType;
+using AbstractOp = AIEBaseInstrInfo::AbstractOp;
+using OptionalOp = std::optional<AbstractOp>;
+
+bool getGlobalValue(const MachineInstr *MI,
+                    SmallPtrSet<const Value *, 4> &GVSet,
+                    MachineRegisterInfo &MRI) {
+
+  MI = getDefIgnoringCopiesAndBitcasts(MI->getOperand(1).getReg(), MRI);
+
+  // We need an instruction that explicitly loads a global.
+  if (MI->getNumOperands() != 2)
```

Comment:
Nit: I find the comment a bit confusing, as the instruction here is not a load, but typically a MOV immediate. Maybe change the comment, or explicitly verify that MI is a MOV immediate?